src/Commands/GenerateSitemap.php

```diff
@@ -0,0 +1,148 @@
+            ->filter(function ($taxonomy) use ($site) {
+                $terms = $taxonomy->queryTerms()
+                    ->where('site', $site->handle())
+                    ->get();
+
+                return $terms
+                    ->filter(function ($term) {
+                        return view()->exists($term->template());
+                    })->isNotEmpty();
+            });
+
+        foreach ($taxonomies as $taxonomy) {
+            /* @var Taxonomy $taxonomy */
+            $sitemap .= $this->addSitemap('sitemap_statamic_taxonomy_' . $site->handle() . '_' . $taxonomy->handle() . '.xml');
+            $this->generateTaxonomySitemap($site, $taxonomy);
+        }
+
+        $sitemap .= '</sitemapindex>';
+        file_put_contents(public_path('sitemap_statamic_' . $site->handle() . '.xml'), $sitemap);
+    }
+
+    protected function generateCollectionSitemap(Site $site, Collection $collection) : void
+    {
+        $sitemap = '<?xml version="1.0" encoding="UTF-8"?>';
+        $sitemap .= '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">';
+        $entries = $collection->queryEntries()->where('site', $site->handle())->whereStatus('published')->get();
+
+        foreach ($entries as $entry) {
+            /* @var Entry $entry */
+            $sitemap .= $this->addUrl($entry->url(), $entry->lastModified());
+        }
+
+        $sitemap .= '</urlset>';
+        file_put_contents(public_path('sitemap_statamic_collection_' . $site->handle() . '_' . $collection->handle() . '.xml'), $sitemap);
+    }
+
+    protected function generateTaxonomySitemap(Site $site, Taxonomy $taxonomy) : void
+    {
+        $sitemap = '<?xml version="1.0" encoding="UTF-8"?>';
+        $sitemap .= '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">';
+        $sitemap .= $this->generateTermSitemap($site);
+        $sitemap .= '</urlset>';
+
+        file_put_contents(public_path('sitemap_statamic_taxonomy_' . $site->handle() . '_' . $taxonomy->handle() . '.xml'), $sitemap);
+    }
+
+    protected function generateTermSitemap(Site $site) : string
+    {
+        $sitemap = '';
+        $terms = $this->publishedTerms($site);
+
+        foreach ($terms as $term) {
+            /* @var LocalizedTerm $term */
+            $sitemap .= $this->addUrl($term->absoluteUrl(), $term->lastModified());
+        }
+
+        return $sitemap;
+    }
+
+    protected function publishedTerms(Site $site)
+    {
+        return TaxonomyFacade::all()
+            ->flatMap(fn ($taxonomy) => $taxonomy->queryTerms()->where('site', $site->handle())->get())
+            ->filter
+            ->published()
+            ->filter(function ($term) {
+                return view()->exists($term->template());
+            });
+    }
+
+    protected function addSitemap($url = null) : string
+    {
+        $sitemap = '<sitemap>';
+
+        if ($url ?? false) {
```

Comment:
`$url` is a method param so always there

```suggestion
        if ($url) {
```